### DISCUSSION TITLE
Enhanced Deep Residual Networks for Single Image Super-Resolution

### DISCUSSION THREAD

**Discussion by harshraj22** [harshraj22/DeepLearning#10] [Paper Review]
# Enhanced Deep Residual Networks for Single Image Super-Resolution [:notebook:](https://openaccess.thecvf.com/content_cvpr_2017_workshops/w12/papers/Lim_Enhanced_Deep_Residual_CVPR_2017_paper.pdf)

<ins>CVPRW 2017</ins>

## Architectural details:

- Built on the top of generator of pre-existing SRResNet model. The authors reduce the number of parameters by removing the unnecessary modules, without affecting the accuracy much.
- The architectural blocks are almost similar for cases when the image is upsampled by x2, x3, x4. This allows them to share weights for most blocks from a pretrained network across the sampling.
- They use L1 loss as the loss function for training the networks.
- The benchmark their results on various dataset, and achieve good results on Div2k dataset.
- They consider the following 2 metric for comparison of their result:
       - PSNR
       - [SSIM](https://medium.com/srm-mic/all-about-structural-similarity-index-ssim-theory-code-in-pytorch-6551b455541e)

![Global Reconstruction](https://github.com/harshraj22/DeepLearning/blob/main/img/EDSR.png?raw=true)
![Global Reconstruction](https://github.com/harshraj22/DeepLearning/blob/main/img/MDSR.png?raw=true)

## Dataset Details:
- The model was trained on DIV2K dataset containing 800 images on train split.
- For training, they use the RGB input patches of size 48×48 from LR image with the corresponding HR patches.
- They augment the training data with random horizontal flips and 90 rotations and pre-process all the images by subtracting the mean RGB value of the DIV2K dataset
